camayoc/tests/qpc/cli/test_sources.py

```diff
@@ -606,6 +622,140 @@ def test_add_with_disable_ssl_negative(
     assert qpc_source_add.exitstatus == exitstatus
 
 
+@pytest.mark.parametrize('source_type, hosts, exclude_hosts',
+                         VALID_SOURCE_TYPE_HOSTS_WITH_EXCLUDE_HOSTS)
+def test_add_with_exclude_hosts(
+        isolated_filesystem, qpc_server_config, hosts, exclude_hosts,
+        source_type):
+    """Add a source with cred and hosts and exclude a host.
+
+    :id: a54567f9-c26c-45f6-9054-5bf419a791fd
+    :description: Add a source entry providing the ``--name``, ``--cred``,
+        ``--hosts``, and ``--exclude-hosts`` options.
+    :steps: Run ``qpc source add --name <name> --cred <cred>
+        --hosts <hosts> --exclude-hosts <excludedhosts> --type <type>``
+    :expectedresults: A new source entry is created with the data provided as
+        input.
+    """
+    cred_name = utils.uuid4()
+    name = utils.uuid4()
+    port = default_port_for_source(source_type)
+    cred_add_and_check(
+        {
+            'name': cred_name,
+            'username': utils.uuid4(),
+            'password': None,
+            'type': source_type,
+        },
+        [(CONNECTION_PASSWORD_INPUT, utils.uuid4())],
+    )
+
+    qpc_source_add = pexpect.spawn(
```

Comment:
Do you think that using source_add_and_check here would be work? https://github.com/quipucords/camayoc/blob/master/camayoc/tests/qpc/cli/utils.py#L185